client/me/help/help-contact-form/index.jsx

```diff
@@ -61,6 +70,22 @@ module.exports = React.createClass( {
 		};
 	},
 
+	componentWillReceiveProps: function( nextProps ) {
+		if ( isEqual( nextProps.valueLink.value, this.state ) ) {
+			return;
+		}
+
+		this.setState( nextProps.valueLink.value );
+	},
+
+	componentDidUpdate: function( prevProps, prevState ) {
+		if ( isEqual( prevState, this.state ) ) {
+			return;
+		}
+
+		this.props.valueLink.requestChange( Object.assign( {}, this.state ) );
```

Comment:
The cloning needs to happen somewhere to ensure that the state object doesn't accidentally get modified and I'd rather it be here since it is the earliest point of use.